release notes/v0.53.0.md

```diff
@@ -0,0 +1,295 @@
+
+As part of the implementation, two common patterns broken in the ecosystem became apparent.
+
+One is arguably a developer experience improvement, and the other is a consequence of the previous implementation.
+
+#### Mixing CommonJS and ESM
+
+Previously, k6 used a transpiler (Babel) internally to transpile ESM syntax to CommonJS. That led to all code always being CommonJS, and if you had CommonJS next to it, Babel would not complain.
+
+As k6 (or the underlying JS VM implementation) did not understand ESM in itself and that CommonJS is a 100% during execution feature, this was not easy to detect or prevent.
+
+We added a  [warning](https://github.com/grafana/k6/pull/3807) in v0.52.0 to give users time for migration.
+
+To fix this - all you need is to stick to either CommonJS or ESM within each file.
+
+<details>
+  <summary>Code examples and proposed changes</summary>
+
+```javascript
+import { sleep } from "k6";
+
+module.exports.default = func() { ...}
+```
+
+In the example above both ESM and CommonJS are used in the same file.
+
+Either replace
+```javascript
+module.exports.default = func() {}
+```
+
+with the ESM syntax
+
+```javascript
+export default func() {}
+```
+
+or replace
+
+```javascript
+import { sleep } from "k6";
+```
+ with the CommonJS:
+
+```javascript
+const sleep = require("k6").sleep;
+```
+
+</details>
+
+#### Imported identifier that can't be resolved are now errors
+
+Previously to this if you were using the ESM syntax but imported identifier `foo` but the exporting file didn't export it - there will be no error.
+
+bar.js:
+```javascript
+export const notfoo = 5;
+```
+main.js
+
+```javascript
+import { foo } from "./bar.js"
+export default function () {
+    foo.bar(); // throws exception here
+}
+```
+It would not error out, but when it is accessed, there will be an exception as it will be `undefined`.
+
+With native ESM support, that is an error as defined by the specification and will occur sooner.
+
+This arguably improves UX/DX, but we have reports that some users have imports like this but do not use them. So, they wouldn't be getting exceptions, but they would now get errors.
+
+The solution, in this case, is to stop importing the not exported identifiers.
+
+### No more "magic" URL resolution
+
+For a long time, k6 has supported special _magic_ URLs that aren't really that.
+
+Those were URLs without a scheme, that:
+1. started with `github.com` that if pasted in a browser will not open you the file. Their appeal was that you can more easily write them by hand if you know the path within a github repo.
+2. started with `cdnjs.com` that if pasted in a browser will open an web page with all the versions of the library. The appeal here is that you will get the latest version.
+
+Both of them had problems though.
+
+The github ones seemed to have never be used by users, likely because you need to guess what the path should look like, and you can always just go get a real URL to the raw file.
```

Comment:
```suggestion
The GitHub ones seemed to have never been used by users, likely because you need to guess what the path should look like, and you can always just go get a real URL to the raw file.
```